### PR TITLE
[CR] [WIP] Extend inv_map_splice to show items within radius

### PR DESCRIPTION
**Pull request by mugling**
Initial implementation of #13955 

Should adjacent items appear under the category `GROUND` or possibly `NEARBY` and the case of the latter do we also need `NEARBY VEHICLE`?
- [X] Implement radius argument
- [ ] Fix inventory letters (existing implementation is broken anyway)
- [X] Restore ability to select items from vehicles

### CODE REVIEW

**Comment by remyroy** [comment]
This has implications with #13907 my current work on implementing the use of nearby container for pouring liquid from crafting.

File: src/inventory_ui.cpp [hunk 1]
```diff
@@ -838,70 +838,70 @@ constexpr char first_invlet = '0';
...
+template<typename Collection, typename Location>
+void pseudo_inv_to_slice( Collection here, Location pos, item_filter filter,
                           pseudo_inventory &item_stacks, indexed_invslice &result_slice,
-                          std::vector<item *> &selectables, char &cur_invlet )
+                          std::vector<std::pair<Location, item *>> &selectables, char &cur_invlet )
 {
+    for (auto& candidate : here) {
+        if (filter(candidate)) {
+            auto stack = std::find_if(item_stacks.begin(), item_stacks.end(),
+                                      [&](const std::list<item>& e) { return candidate.stacks_with(e.back()); });
```
Comment (Coolthulhu) [inline src/inventory_ui.cpp]: Lambdas generally look the most readable when they are written like functions, ie. first line for the declaration - `[]() {` - then the body, then on the last line the closing brace.

**Comment by Coolthulhu** [comment]
> map_stack and vehicle_stack use tripoint and point respectively

Vehicle stacks doesn't keep the actual item location. Instead, it keeps the vehicle location and vehicle part (relative to vehicle) location. It could possibly just keep the part index instead.

**Comment by mugling** [comment]
> Vehicle stacks doesn't keep the actual item location

Doesn't this need to be stored for the call to `item_location::on_vehicle`?

File: src/inventory_ui.cpp [hunk 2]
```diff
@@ -953,10 +936,10 @@ item_location game::inv_map_splice(
             const size_t index = (size_t)(ch - first_invlet);
```
Comment (mugling) [inline src/inventory_ui.cpp]: This looks like it's already broken

**Comment by mugling** [comment]
Consistently segfaults when two different items (that don't stack) are present on the same tile. Anyone see anything obvious?

**Comment by mugling** [comment]
Change of approach, see #13986

**Pull request closed without merging**